src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -412,11 +426,11 @@ private static function parseUrl(string $url, array $query = [], array $allowedS
 
             if (false !== strpos($parts[$part], '%')) {
                 // https://tools.ietf.org/html/rfc3986#section-2.3
-                $parts[$part] = preg_replace_callback('/%(?:2[DE]|3[0-9]|[46][1-9A-F]|5F|[57][0-9A]|7E)++/i', function ($m) { return rawurldecode($m[0]); }, $parts[$part]);
+                $parts[$part] = preg_replace_callback('/%(?:2[DE]|3[0-9]|[46][1-9A-F]|5F|[57][0-9A]|7E)++/i', static function ($m) { return rawurldecode($m[0]); }, $parts[$part]);
```

Comment:
just for the comment: using `static` here provides no practical benefit AFAIK (same for all other places where it's been adde)

Reply:
Ok, I didn't know this practice, I saw it by reading other part of the component (`or Symfony? I don't remember exactly`), so I though it would be needed to do it here.
That said, I reverted my changes.

Reply:
For the record, using `static` removes some circular references when a closure is assigned to a property of the same class where it has been created.
Here, these closures are not assigned so the issue doesn't apply.

Reply:
Thanks.